src/HelperTraits/GTINMigrationUtilities.php

```diff
@@ -92,63 +94,94 @@ protected function options() {
+	 * @param string     $gtin
 	 * @return string
 	 */
-	protected function error_gtin_invalid( \WC_Product $product, string $gtin ) {
+	protected function error_gtin_invalid( WC_Product $product, string $gtin ): string {
 		return sprintf( 'GTIN [ %s ] has been skipped for Product ID: %s - %s. Invalid GTIN was found.', $gtin, $product->get_id(), $product->get_name() );
 	}
 
 	/**
 	 * Gets the message when the GTIN is already in the Product Inventory
 	 *
-	 * @param \WC_Product $product
+	 * @param WC_Product $product
 	 * @return string
 	 */
-	protected function error_gtin_already_set( \WC_Product $product ) {
+	protected function error_gtin_already_set( WC_Product $product ): string {
 		return sprintf( 'GTIN has been skipped for Product ID: %s - %s. GTIN was found in Product Inventory tab.', $product->get_id(), $product->get_name() );
 	}
 
 	/**
 	 * Gets the message when the GTIN is not found.
 	 *
-	 * @param \WC_Product $product
+	 * @param WC_Product $product
 	 * @return string
 	 */
-	protected function error_gtin_not_found( \WC_Product $product ) {
+	protected function error_gtin_not_found( WC_Product $product ): string {
 		return sprintf( 'GTIN has been skipped for Product ID: %s - %s. No GTIN was found', $product->get_id(), $product->get_name() );
 	}
 
 	/**
 	 * Gets the message when the GTIN had an error when saving.
 	 *
-	 * @param \WC_Product $product
-	 * @param string      $gtin
-	 * @param Exception   $e
+	 * @param WC_Product $product
+	 * @param string     $gtin
+	 * @param Exception  $e
 	 *
 	 * @return string
 	 */
-	protected function error_gtin_not_saved( \WC_Product $product, string $gtin, Exception $e ) {
+	protected function error_gtin_not_saved( WC_Product $product, string $gtin, Exception $e ): string {
 		return sprintf( 'GTIN [ %s ] for Product ID: %s - %s has an error - %s', $gtin, $product->get_id(), $product->get_name(), $e->getMessage() );
 	}
 
 	/**
 	 * Gets the message when the GTIN is successfully migrated.
 	 *
-	 * @param \WC_Product $product
-	 * @param string      $gtin
+	 * @param WC_Product $product
+	 * @param string     $gtin
 	 *
 	 * @return string
 	 */
-	protected function successful_migrated_gtin( \WC_Product $product, string $gtin ) {
+	protected function successful_migrated_gtin( WC_Product $product, string $gtin ): string {
 		return sprintf( 'GTIN [ %s ] has been migrated for Product ID: %s - %s', $gtin, $product->get_id(), $product->get_name() );
 	}
+
+	/**
+	 * Gets the GTIN value
+	 *
+	 * @param WC_Product $product The product
+	 * @return string|null
+	 */
+	protected function get_gtin( WC_Product $product ): ?string {
+		$gtin = $this->attribute_manager->get_value( $product, 'gtin' );
+
+		if ( ! $gtin ) {
+			$gtin = $this->get_yoast_gtin( $product );
+		}
+
+		return $gtin;
+	}
+
+	/**
+	 * Gets the GTIN value from YOAST SEO
+	 *
+	 * @param WC_Product $product The product
+	 * @return string|null
+	 */
+	protected function get_yoast_gtin( WC_Product $product ): ?string {
```

Comment:
So just add a filter in `get_gtin` so we don't need to add anything in regards to Yoast here. Then in the class  `YoastWooCommerceSeo`  we can hook into the filter for migration and if it has a value to return it can do so.

Reply:
Hey @mikkamp Thanks for this suggestion. I 100% agree.

Here is the adjustment https://github.com/woocommerce/google-listings-and-ads/pull/2677/commits/356577e875a20d9dd5f93e4e5a98817e874620ea